rtcclient/client.py

```diff
@@ -77,56 +79,24 @@ def get_rtc_obj(self):
         return self
 
     def _get_headers(self):
-        if self.jazz is True:
-            _allow_redirects = True
-        else:
-            _allow_redirects = False
-
+        self.http = httplib2.Http()
         _headers = {"Content-Type": self.CONTENT_XML}
-        resp = self.get(self.url + "/authenticated/identity",
-                        verify=False,
-                        headers=_headers,
-                        proxies=self.proxies,
-                        allow_redirects=_allow_redirects)
-
         _headers["Content-Type"] = self.CONTENT_URL_ENCODED
-        if resp.headers.get("set-cookie") is not None:
-            _headers["Cookie"] = resp.headers.get("set-cookie")
-
-        credentials = urlencode({"j_username": self.username,
-                                 "j_password": self.password})
-
-        resp = self.post(self.url + "/authenticated/j_security_check",
-                         data=credentials,
-                         verify=False,
-                         headers=_headers,
-                         proxies=self.proxies,
-                         allow_redirects=_allow_redirects)
-
-        # authfailed
-        authfailed = resp.headers.get("x-com-ibm-team-repository-web-auth-msg")
-        if authfailed == "authfailed":
-            raise exception.RTCException("Authentication Failed: "
-                                         "Invalid username or password")
-
-        # fix issue #68
-        if not _allow_redirects:
-            if resp.headers.get("set-cookie") is not None:
-                _headers["Cookie"] = resp.headers.get("set-cookie")
-
-        resp = self.get(self.url + "/authenticated/identity",
-                        verify=False,
-                        headers=_headers,
-                        proxies=self.proxies,
-                        allow_redirects=_allow_redirects)
-
-        # fix issue #68
-        if not _allow_redirects:
-            _headers["Cookie"] += "; " + resp.headers.get("set-cookie")
-        else:
-            _headers["Cookie"] = resp.headers.get("set-cookie")
-
         _headers["Accept"] = self.CONTENT_XML
+        _headers["OSLC-Core-Version"] = self.OSLC_CORE_VERSION
+        
+        # Start the authentication via j_security_check page
+        resp, content = self.http.request(self.url + '/web/j_security_check', 'POST', headers=_headers,
+                                          body=urllib.urlencode({'j_username': self.username, 'j_password': self.password}))
+        
+        # Save the new cookie returned here
+        _headers["Cookie"] = resp["set-cookie"]
```

Comment:
For failed logins, IIRC field `set-cookie` is `None`.

We need to handle failed logins first before we access `set-cookie`.